inference/inference-tfjs/src/jsMain/kotlin/io.kinference.tfjs/operators/ml/ZipMap.kt

```diff
@@ -0,0 +1,116 @@
+import io.kinference.tfjs.TFJSData
+import io.kinference.tfjs.data.map.TFJSMap
+import io.kinference.tfjs.data.seq.TFJSSequence
+import io.kinference.tfjs.data.tensors.TFJSTensor
+import io.kinference.tfjs.data.tensors.asTensor
+import io.kinference.types.TensorShape
+import io.kinference.types.ValueTypeInfo
+
+sealed class ZipMap(
+    name: String,
+    info: OperatorInfo,
+    attributes: Map<String, Attribute<Any>>,
+    inputs: List<String>,
+    outputs: List<String>
+) : Operator<TFJSTensor, TFJSSequence>(name, info, attributes, inputs, outputs) {
+    companion object {
+        private val DEFAULT_VERSION = VersionInfo(sinceVersion = 1)
+
+        operator fun invoke(name: String, version: Int?, attributes: Map<String, Attribute<Any>>, inputs: List<String>, outputs: List<String>): ZipMap {
+            return when (version ?: DEFAULT_VERSION.sinceVersion) {
+                in ZipMapVer1.VERSION.asRange() -> ZipMapVer1(name, attributes, inputs, outputs)
+                else -> error("Unsupported version of ZipMap operator: $version")
+            }
+        }
+    }
+}
+
+
+class ZipMapVer1(
+    name: String,
+    attributes: Map<String, Attribute<Any>>,
+    inputs: List<String>,
+    outputs: List<String>
+) : ZipMap(name, INFO, attributes, inputs, outputs) {
+    companion object {
+        private val OUT_TYPE_CONSTRAINTS = setOf(TensorProto.DataType.INT64, TensorProto.DataType.STRING, TensorProto.DataType.FLOAT)
+
+        private val INPUTS_INFO = listOf(
+            IOInfo(0, setOf(TensorProto.DataType.FLOAT), "X", optional = false)
+        )
+
+        private val OUTPUTS_INFO = listOf(
+            IOInfo(0, OUT_TYPE_CONSTRAINTS, "Z", optional = false)
+        )
+
+        private val ATTRIBUTES_INFO = listOf(
+            AttributeInfo("classlabels_int64s", setOf(AttributeProto.AttributeType.INTS), required = false),
+            AttributeInfo("classlabels_strings", setOf(AttributeProto.AttributeType.STRINGS), required = false)
+        )
+
+        internal val VERSION = VersionInfo(sinceVersion = 1)
+        private val INFO = OperatorInfo("ZipMap", ATTRIBUTES_INFO, INPUTS_INFO, OUTPUTS_INFO, VERSION, domain = "ai.onnx.ml")
+
+        private fun <T : Any> NumberNDArrayTFJS.asSeqWithLabels(labels: Labels<T>, mapInfo: ValueTypeInfo.MapTypeInfo): TFJSSequence {
+            val seqInfo = ValueTypeInfo.SequenceTypeInfo(mapInfo)
+            val rows = if (rank == 1) 1 else shape[0]
+            val columns = shape.last()
+
+            val inputArray = this.dataFloat()
+            var offset = 0
+            return TFJSSequence("Z", seqInfo, rows) {
+                val map = HashMap<T, TFJSData<*>>(columns)
+                repeat(columns) { col ->
+                    val value = inputArray[offset + col]
+                    val tensor = NDArrayTFJS.floatScalar(value).asTensor()
+                    map[labels[it]] = tensor
+                }
+                offset += columns
```

Comment:
I think we don't need some `offset` counter outside the loop. Use row index from `TFJSSequence` constructor